### PR TITLE
feat: add process identifiers to document metadata

### PR DESCRIPTION
**Pull request by chillleader**
## Description

This PR will add process definition ID and process instance key as optional identifiers to the document metadata.

Adding these values will allow to use them for authorization checks in the future.

**⚠️ This PR is part of the PR train that needs to follow a specific order when merging ⚠️**

@mo-okocha @marcosgvieira Please merge the PRs in this specific order to avoid merge conflicts:

- https://github.com/camunda/camunda/pull/26336
- https://github.com/camunda/camunda/pull/26340
- https://github.com/camunda/camunda/pull/26351

The branches of each PR are pointed towards the branch of the previous PR - please wait for the previous PR to be merged and make sure the target branch of the second PR is changed to main. Only after that you can merge the next PR.

**This PR goes third.**

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes: #24847

### CODE REVIEW

**Comment by chillleader** [comment]
**⚠️ This PR is part of the PR train that needs to follow a specific order when merging ⚠️**

@mo-okocha @marcosgvieira Please merge the PRs in this specific order to avoid merge conflicts:

- https://github.com/camunda/camunda/pull/26336
- https://github.com/camunda/camunda/pull/26340
- https://github.com/camunda/camunda/pull/26351

The branches of each PR are pointed towards the branch of the previous PR - please wait for the previous PR to be merged and make sure the target branch of the second PR is changed to main. Only after that you can merge the next PR.

**This PR goes third and last.**

**Review by mo-okocha — APPROVED** [review]

**Pull request merged**
